libraries/chain/include/eosio/chain/log_catalog.hpp

```diff
@@ -217,23 +215,24 @@ struct log_catalog {
       bfs::path new_path = retained_dir / buf;
       rename_bundle(dir / name, new_path);
       size_type items_to_erase = 0;
-      collection.emplace(start_block_num, mapped_type{end_block_num, new_path});
+      collection.emplace(start_block_num, mapped_type{end_block_num, std::move(new_path)});
       if (collection.size() >= max_retained_files) {
          items_to_erase =
             max_retained_files > 0 ? collection.size() - max_retained_files : collection.size();
+         auto end = std::next( collection.begin(), items_to_erase);
```

Comment:
Minor comment - I think `last` would be a better name than `end` because I initially thought you were iterating till `end()` - reading too fast!